addons/pkg-maven/common/src/main/java/org/commonjava/indy/pkg/maven/content/MavenMetadataGenerator.java

```diff
@@ -1052,13 +1057,14 @@ private boolean writeSnapshotMetadata( final ArtifactPathInfo info, final List<S
             coordMap.put( GROUP_ID, info.getGroupId() );
             coordMap.put( VERSION, info.getReleaseVersion() + LOCAL_SNAPSHOT_VERSION_PART );
 
-            final String lastUpdated = SnapshotUtils.generateUpdateTimestamp( SnapshotUtils.getCurrentTimestamp() );
+            final String lastUpdated = generateUpdateTimestamp( SnapshotUtils.getCurrentTimestamp() );
 
             doc.appendChild( doc.createElementNS( doc.getNamespaceURI(), "metadata" ) );
             xml.createElement( doc.getDocumentElement(), null, coordMap );
 
             xml.createElement( doc, "versioning", Collections.singletonMap( LAST_UPDATED, lastUpdated ) );
```

Comment:
I'm not sure if this one also needs to change. I think this <lastUpdated> tag should be same with the last snapshot file's <update> tag, to determine the right file. But seems it is still the current timestamp.

Reply:
Really good catch! I update the pr. pls look again.

Reply:
I also simplify it a bit. We do not need to worry about null-timestamp. it should never happen. if that happens, let it fail.

Reply:
Ok. LGTM